service/src/main/java/org/cbioportal/service/impl/GenesetDataServiceImpl.java

```diff
@@ -71,13 +71,10 @@
         List<Integer> internalSampleIds = Arrays.stream(commaSeparatedSampleIdsOfGeneticProfile.split(","))
             .mapToInt(Integer::parseInt).boxed().collect(Collectors.toList());
 
-        List<Sample> samples;
-        if (sampleIds == null) {
-            samples = sampleService.getSamplesByInternalIds(internalSampleIds);
-        } else {
-            List<String> studyIds = new ArrayList<>();
-            sampleIds.forEach(s -> studyIds.add(molecularProfile.getCancerStudyIdentifier()));
-            samples = sampleService.fetchSamples(studyIds, sampleIds, "ID");
+        List<Sample> samples = sampleService.getSamplesByInternalIds(internalSampleIds);
```

Comment:
getSamplesByInternalIds returns `UnsupportedOperationException`
https://github.com/cBioPortal/cbioportal/pull/6440/files#diff-95d528fcc25c8f87f6828986b804f6eaR124

Reply:
correct, the internalIds are based on MySQL, and not accessible from parquet data.

Reply:
If this is based on MySQL then why are these changes required?

Reply:
Thanks for asking that question. Later in this method, `int indexOfSampleId = internalSampleIds.indexOf(sample.getInternalId());` it expects to find internalId of sample, which is MySQL specific. The implemented fetchSamples cannot retrieve internalIds from txt data and it would be duplicate effort to go back to MySQL just to fetch the internalId. The main difference I saw in the `if (sampleIds==null)/else` was that it was filtering by smapleIds if provided, so the fetchSample call was removed and filtering by sampleId was added.

Reply:
In [SampleServiceImpl](https://github.com/cBioPortal/cbioportal/blob/4fb750f4b3c6adeb6bf89b263568c4f304c0310e/service/src/main/java/org/cbioportal/service/impl/SampleServiceImpl.java#L189), some code also depend on internalId, however it doesn't affect this endpoint (the line is only called with DETAILED projection). For sample-counts endpoint (PR to be created) I had to make changes so that the code doesn't depend on `internalId` and instead on `sampleId`. If this is a hard dependency (i.e. different samples can have same `sampleId` but different `internalId`), then it would not be appropriate to implement fetchSamples Parquet.

Reply:
Unfortunately, only internal (MySQL) sample ids are unique across studies.  The stable id, like TCGA-A1-A0SB-01 is not unique across studies as the sample could be combined with samples from other studies into a new "merged" study.  We could consider the combined study-id/sample-id to be unique.

Reply:
Thanks for the confirmation. I added code to check both studyId/sampleId